beacon-chain/db/filesystem/data_column.go

```diff
@@ -0,0 +1,966 @@
+package filesystem
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/async"
+	"github.com/prysmaticlabs/prysm/v5/async/event"
+	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v5/io/file"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/time/slots"
+	"github.com/spf13/afero"
+)
+
+const (
+	version                                    int = 0x01
+	versionSize                                    = 1           //bytes
+	maxSszEncodedDataColumnSidecarSize             = 536_870_912 // 2**(4*8) / 8 bytes
+	encodedSszEncodedDataColumnSidecarSizeSize     = 4           // bytes (size of the endoded size of the SSZ encoded data column sidecar)
+	mandatoryNumberOfColumns                       = 128         // 2**7
+	limit                                          = mandatoryNumberOfColumns
+	headerSize                                     = versionSize + encodedSszEncodedDataColumnSidecarSizeSize + mandatoryNumberOfColumns
+	dataColumnsFileExtension                       = "sszs"
+	prunePeriod                                    = 1 * time.Minute
+)
+
+var (
+	ErrEncodedDataColumnSidecarTooLarge     = errors.New("encoded data columns sidecar too large")
+	ErrWrongNumberOfColumns                 = errors.New("wrong number of data columns")
+	ErrDataColumnIndexTooLarge              = errors.New("data column index too large")
+	ErrWrongBytesWritten                    = errors.New("wrong number of bytes written")
+	ErrWrongBytesVersionRead                = errors.New("wrong number of bytes version read")
+	ErrWrongVersion                         = errors.New("wrong version")
+	ErrWrongBytesDataColumnSidecarSizeRead  = errors.New("wrong number of bytes data column sidecar size read")
+	ErrWrongBytesIndicesRead                = errors.New("wrong number of bytes indices read")
+	ErrWrongFileSize                        = errors.New("wrong file size")
+	ErrTooManyDataColumns                   = errors.New("too many data columns")
+	ErrWrongSszEncodedDataColumnSidecarSize = errors.New("wrong SSZ encoded data column sidecar size")
+	ErrDataColumnSidecarsFromDifferentSlots = errors.New("data column sidecars from different slots")
```

Comment:
I think most or all of these variables can be un-exported.

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/15061/commits/c6e8c1a380068f2687e19a2e9c1e78265bc02463.